warehouse/ingest-core/src/test/java/datawave/ingest/mapreduce/handler/shard/ShardedDataTypeHandlerTest.java

```diff
@@ -1,17 +1,176 @@
+        
+        conf.set(ShardedDataTypeHandler.METADATA_TABLE_NAME, TableName.METADATA);
+        conf.set(ShardedDataTypeHandler.NUM_SHARDS, Integer.toString(NUM_SHARDS));
+        conf.set(ShardedDataTypeHandler.SHARDED_TNAMES, TableName.SHARD + "," + TableName.ERROR_SHARD);
+        conf.set(ShardedDataTypeHandler.SHARD_TNAME, TableName.SHARD);
+        conf.set(ShardedDataTypeHandler.SHARD_LPRIORITY, "30");
+        conf.set(TableName.SHARD + TableConfigHelper.TABLE_CONFIG_CLASS_SUFFIX, ShardTableConfigHelper.class.getName());
+        conf.set(ShardedDataTypeHandler.SHARD_GIDX_TNAME, TableName.SHARD_INDEX);
+        conf.set(ShardedDataTypeHandler.SHARD_GIDX_LPRIORITY, "30");
+        conf.set(TableName.SHARD_INDEX + TableConfigHelper.TABLE_CONFIG_CLASS_SUFFIX, ShardTableConfigHelper.class.getName());
+        conf.set(ShardedDataTypeHandler.SHARD_GRIDX_TNAME, TableName.SHARD_RINDEX);
+        conf.set(ShardedDataTypeHandler.SHARD_GRIDX_LPRIORITY, "30");
     }
+    
+    @Before
+    public void setUp() throws Exception {
+        configuration = new Configuration();
+        setupConfiguration(configuration);
+        handler = new AbstractColumnBasedHandler<>();
+        handler.setup(new TaskAttemptContextImpl(configuration, new TaskAttemptID()));
+        handler.setShardIndexTableName(new Text("shardIndex"));
+        handler.setShardReverseIndexTableName(new Text("shardReverseIndex"));
+        
+        dataTypeHandler = new AbstractColumnBasedHandler<>();
+        dataTypeHandler.setup(new TaskAttemptContextImpl(configuration, new TaskAttemptID()));
+        
+        ingestHelper = new TestIngestHelper();
+        ingestHelper.setup(configuration);
+        
+        maskedFieldHelper = new TestMaskedHelper();
+        maskedFieldHelper.setup(configuration);
+    }
+    
+    @Test
+    public void testCreateTermIndex() {
+        Type dataType = new Type(DATA_TYPE_NAME, TestIngestHelper.class, null, null, 10, null);
+        String entry = "testingtesting";
+        RawRecordContainer record = new RawRecordContainerImpl();
+        record.setDataType(dataType);
+        record.setRawFileName("data_" + 0 + ".dat");
+        record.setRawRecordNumber(1);
+        record.setRawData(entry.getBytes(StandardCharsets.UTF_8));
+        
+        Uid.List uid = Uid.List.newBuilder().setIGNORE(false).setCOUNT(1).addUID("d8zay2.-3pnndm.-anolok").build();
+        byte[] visibility = new byte[] {65, 76, 76};
+        byte[] shardId = new byte[] {50, 48, 48, 48, 48, 49, 48, 49, 95, 54, 57};
+        
+        Multimap<BulkIngestKey,Value> termIndex = handler.createTermIndexColumn(record, "FIELD_NAME", "FIELD_VALUE", visibility, null, null, shardId,
+                        handler.getShardIndexTableName(), new Value(uid.toByteArray()), Direction.FORWARD);
+        
+        assertTrue(termIndex.size() == 1);
+    }
+    
+    @Test
+    public void testCreateTermReverseIndex() {
```

Comment:
Nice test!

Reply:
Thanks! Going to iterate tests on top of this.